logdetective/logdetective.py

```diff
@@ -92,15 +93,21 @@ def main():
     if args.no_stream:
         stream = False
     response = process_log(log_summary, model, stream)
+    probs = []
     print("Explanation:")
     if args.no_stream:
         print(response["choices"][0]["text"])
+        probs = response["choices"][0]["logprobs"]["top_logprobs"]
     else:
         # Stream the output
         for chunk in response:
+            if isinstance(chunk["choices"][0]["logprobs"], dict):
+                probs.extend(chunk["choices"][0]["logprobs"]["top_logprobs"])
             delta = chunk['choices'][0]['text']
             print(delta, end='', flush=True)
-        print()
+    certainty = compute_certainty(probs)
+
+    print(f"\nResponse certainty: {certainty:.2f}%\n")
```

Comment:
I think we can go low as 1f or even 0f.

Reply:
I'm sure we can, although I kind of like the two decimal cut off.

Reply:
Then leave it there. We will ask users later how they like it.